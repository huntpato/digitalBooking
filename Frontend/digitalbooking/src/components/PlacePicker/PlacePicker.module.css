.container{
    width: 100%;
}

.input{
    border-radius: 5px;
    height:25px;
    padding: 15px;
    width: 98%;
}

.containerLocations{
    background-color:white;
    border-bottom-left-radius: 8px;
    border-bottom-right-radius: 8px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    margin-top: 2px;
    position:absolute;
    width: 24.5%;
}

.hide{
    display:none;
}

@media (max-width: 768px){
    .containerLocations{
        width: 27%;
    }
}

@media (max-width: 576px){


    .input{
        height:35px;
        width: 100%;
        margin-bottom: 10px;
    }

    .containerLocations{
        width: 90%;
        z-index: 2;
    }
}

@media (max-width: 480px){


    .input{
        height:35px;
        width: 100%;
        margin-bottom: 10px;
    }

    .containerLocations{
        margin-top: 0;
        width: 90%;
        z-index: 2;

    }
}