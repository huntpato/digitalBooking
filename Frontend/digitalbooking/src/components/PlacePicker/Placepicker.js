import React, { useContext } from "react";
import { LocationContext } from "../SearchBar/LocationContext";
import styles from "./PlacePicker.module.css";
import useFetch from "use-http";
import { PlaceElement } from "../PlaceElement/PlaceElement";
import { useState } from "react";

export const Placepicker = () => {
  const { setPlace } = useContext(LocationContext);
  const [setOpen] = useState(false);
  const options = {
    headers: {
      "X-Requested-With": "XMLHttpRequest",
    },
  };
  const [search, setSearch] = useState("");
  const [locations, setLocations] = useState([]);
  const url =
    "https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/products/locations";
  const { data = [] } = useFetch(url, options, []);
  const { input, containerLocations, locaciones, container } = styles;
  let matches = [];

  const onSuggestHandler = (text) => {
    setSearch(text);
    setPlace(text);
    setLocations([]);
  };

  const onChangeHandler = (text) => {
    setSearch(text);
    if ((text === "")) {
      setLocations([]);
    } else {
      matches = data.sort().filter((city) => {
        let cityName = city.split(',')[0];
        const regex = new RegExp(`${text}`, "gi");
        return cityName.match(regex);
      });

      if(matches.length === 0){
        matches[0]=`Sin Coincidencias, No se encontraron resulados con ${text}`
      } 
      setLocations(matches);
    }
  };

  return (
    <div className={container}>
      <input
        className={input}
        type="text"
        placeholder="¿A dónde vamos?"
        onChange={(e) => onChangeHandler(e.target.value)}
        value={search}
        onBlur={() => {
          setTimeout(() => {
            setLocations([]);
          }, 200);
        }}
        content="width=device-width, initial-scale=1, maximum-scale=1"
      />
      <div className={containerLocations}>
        <ul className={locaciones}>
          {locations &&
            locations.slice(0, 4).map((city, index) => (
              <div onClick={() => onSuggestHandler(city)}>
                <PlaceElement key={index} city={city} search={search} />
              </div>
            ))}
        </ul>
      </div>
    </div>
  );
};
