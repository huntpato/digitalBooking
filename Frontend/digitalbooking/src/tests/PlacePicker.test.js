import React from 'react';
import { shallow } from 'enzyme';
import { Placepicker } from '../components/PlacePicker/Placepicker'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

global.fetch = jest.fn(()=> Promise.resolve({
    json: ()=>Promise.resolve(
    ["Buenos Aires", "San Francisco"])
}))
  
  jest.mock('react-router-dom', () => ({
    useParams: jest.fn().mockReturnValue({ nifUuid: 'nif123' }),
    useHistory: jest.fn(),
    useContext: jest.fn().mockReturnValue({visibility: true}),
    useLocation:jest.fn().mockReturnValue({
      pathname: "/register",
      search: "",
      hash: "",
      state: null,
      key: "5nvxpbdafa",
    }),
  }));
  
describe('Pruebas en componente <Placepicker />', () => {
    const wrapper= shallow(<Placepicker/>);
   
    test('Comprobar que <Placepicker /> renderiza correctamente.', () => {

        expect( wrapper ).toMatchSnapshot();
    })

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const div = wrapper.find('div').at(0);
        expect(div.hasClass('container')).toBe(true);
    })

    test('Verificar que el componente carga el input', () => {

        const div = wrapper.find('input').at(0);
        expect(div.hasClass('input')).toBe(true);
    })

    
    test('Verificar que el componente carga el placeholder', () => {

        const div = wrapper.find('input').at(0);
        expect(div.prop('placeholder')).toBe("¿A dónde vamos?");
    })
    
    test('Verificar que el componente carga la lista', () => {

        const div = wrapper.find('ul').at(0);
        expect(div.hasClass('locaciones')).toBe(true);
    })

    test('Verificar que el input recibe sugerencias', () => {

        const input = wrapper.find('input');
        input.simulate('change', { target: { value: 'Buenos' } })
       
    })
    
})
