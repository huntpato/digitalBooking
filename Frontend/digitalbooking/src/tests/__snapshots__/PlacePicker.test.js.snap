// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <Placepicker /> Comprobar que <Placepicker /> renderiza correctamente. 1`] = `
<div
  className="container"
>
  <input
    className="input"
    content="width=device-width, initial-scale=1, maximum-scale=1"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="¿A dónde vamos?"
    type="text"
    value=""
  />
  <div
    className="containerLocations"
  >
    <ul
      className="locaciones"
    />
  </div>
</div>
`;
